pkg/splunk/enterprise/configuration.go

```diff
@@ -425,16 +427,35 @@ func getSplunkStatefulSet(client splcommon.ControllerClient, cr splcommon.MetaOb
 	}
 
 	// update statefulset's pod template with common splunk pod config
-	updateSplunkPodTemplateWithConfig(&statefulSet.Spec.Template, cr, spec, instanceType, extraEnv, statefulSetSecret.GetName())
+	updateSplunkPodTemplateWithConfig(client, &statefulSet.Spec.Template, cr, spec, instanceType, extraEnv, statefulSetSecret.GetName())
 
 	// make Splunk Enterprise object the owner
 	statefulSet.SetOwnerReferences(append(statefulSet.GetOwnerReferences(), splcommon.AsOwner(cr, true)))
 
 	return statefulSet, nil
 }
 
+// getConfigMap gets the ConfigMap resource in a given namespace
+func getConfigMap(client splcommon.ControllerClient, namespacedName types.NamespacedName) (corev1.ConfigMap, error) {
+	var configMap corev1.ConfigMap
+	err := client.Get(context.TODO(), namespacedName, &configMap)
+	if err != nil {
+		return configMap, err
+	}
+	return configMap, nil
+}
+
+// getConfigMapResourceVersion gets the Resource version of a configMap
+func getConfigMapResourceVersion(client splcommon.ControllerClient, namespacedName types.NamespacedName) (string, error) {
```

Comment:
Do we need this API since we are extracting just a single field from the configMap?

Reply:
Being it is metatdata info, and using across places, my vote is to have this API

Reply:
I am keeping this API for now. Its always good to have some APIs to get important fields.